draft-ietf-cdni-ci-triggers-rfc8007bis.xml

```diff
@@ -2202,12 +2202,58 @@ uCDN-A                         dCDN-B                         dCDN-C
 	</texttable>
 	</section>
 
+	<section title="URL Type" anchor="url-types">
+	<t>
+	This type is used by uCDN to indicate how to interpret URLs referenced by trigger specs that use URLs, such as 
+		<xref target="urls-spec"/>, <xref target="uri-pattern-match-spec"/>, <xref target="uri-regex-match-spec"/> and <xref target="objectlist-spec"/>.
+	</t>
+	<t>
+	One option for uCDN to use in Trigger Commands is published URLs, that are used by end users.
+	When using this URL type, uCDN MUST provide configuration metadata objects related to these URLs before creating a trigger option referencing these URLs. When this is not the case, dCDN MUST return error code "emeta".
+	dCDN 
+	</t>
+	<t> 
+		When processing published URLs in "preposition" Trigger Action, dCDN MUST invoke processing of metadata objects, it would have involved in content acquisition to satisfy an end user request, e.g. SourceMetaData (see Section 4.2.1 of <xref target="RFC8006"/>.
+	</t>
+
+	<t>
+		Another type of URL in common use is private URL based on cache keys that are dynamically 
+		constructed via lightweight processing of various properties of the HTTP request and/or response. 
+		As an example, an origin might specify a cache key as a value returned in a specific HTTP response header.
+	</t>
+	<t>
+		As an example, uCDN may be interested to use such private URLs in "purge" or "invalidate" 
+		Trigger Action to simplify processing. 
+	</t>
+	<t>	uCDN MUST NOT mix different URL types in same Trigger Command</t>
+	<t>
+		The following URL types are defined by this document and MUST be supported by implementation 
```

Comment:
Re: MUST, I'm not sure that I agree with making this mandatory, and it could become a barrier to the adoption of this specification.  Purging based on the published URL has been a bit of an unofficial standard in the CDN industry, and based on experience with various CDNs, many existing purge implementations are built in this way without support for purging private URLs.  Purging based on private URL would likely result in easier processing, at least in the case of absolute URLs, although could be tricky to support for patterns and regexes given that these CDNs have developed features to improve efficiency of these types of purges based on the published URL, but not the private URL.  It may not be so easy for CDNs with existing purge implementations to reverse course and also support private URL purges without major changes.  For this reason, I would support making the published URL a required default, and allow purging by private URL as an added capability based on the capabilities advertisement.

Reply:
Agreed, added the new FCI capability object and referenced it from the URL Type section
The comment about uCDN separating private and published URLs still stands though, provided that dCDN does support the private URL type